### ISSUE TITLE
Host not updated with Vm info if Create has errors

### ISSUE THREAD

**Issue by kolds** [skytap/docker-machine-driver-skytap#16]
Repro:
Using an environment setup that requires a VPN for SSH to work, run create without specifying skytap-vpn-id.
```
$ docker-machine create -d skytap \
> --skytap-ssh-user docker \
> --skytap-vm-cpus 4 \
> --skytap-vm-ram 4096 \
> --skytap-vm-id 9646772 \
> ko-repro-01
```

Request creates the vm in Skytap and starts the vm, but then fails with:
```
(ko-repro-01) Unable to SSH to target machine to copy public key credentials after retries: dial tcp 10.0.0.1:22: getsockopt: operation timed out
Error creating machine: Error in driver during machine creation: dial tcp 10.0.0.1:22: getsockopt: operation timed out
```

Inspecting the newly created Docker host shows the Vm section has not been setup. 
```
        "Vm": {
            "id": "",
            "name": "",
            "runstate": "",
            "error": null,
            "Interfaces": null,
            "hardware": {}
        },
```
Without Vm.id any attempts to manage the host (start, stop, rm, etc) will fail. 

```
$ docker-machine stop ko-repro-01
Stopping "ko-repro-01"...
json: cannot unmarshal array into Go value of type api.VirtualMachine

$ docker-machine rm ko-repro-01
About to remove ko-repro-01
Are you sure? (y/n): y
Error removing host "ko-repro-01": Resource not found

(ls shows the host was not removed)
$ docker-machine ls
NAME          ACTIVE   DRIVER       STATE     URL                       SWARM   DOCKER    ERRORS
ko-repro-01   -        skytap       Error                                       Unknown   json: cannot unmarshal array into Go value of type api.VirtualMachine
```

The vm can be managed using Skytap, but the status will not be updated in docker-machine.

**Comment by dantjones** [comment]
It looks like the driver should catch this case and not create the VM if a VPN will be needed but not supplied.

**Comment by kolds** [comment]
I did some research into this over the weekend. VPN missing is just one symptom of a bigger issue.

This is what docker machine does when creating a skytap host:
1. Get the config values (SetConfigFromFlags).
2. Run PreCreateCheck. Currently our PreCreateCheck does nothing, so this check always passes.
3. If PreCreateCheck fails, the attempt terminates.
4. Save the Host using values from SetConfigFromFlags. This creates the base host, Vm section contains blank/null values.
5. Run Create. There are several times during the create process that we update Host values. 
6. If Create fails, the attempt terminates. The base host in step 4 still exists, but updates made during step 5 are not saved.
7. Save the Host.
8. Provision docker on the new host.

Any errors during Create will result in updates to Host not being saved. At this point any calls that rely on Vm will fail. The following commands rely on Vm:
GetURL
GetState
Kill
Remove
Restart
Start
Stop

These commands should handle blank Vm to avoid "json: cannot unmarshal array into Go value of type api.VirtualMachine" from being displayed. Also Remove should be able to remove the host from docker-machine when we do not have Vm info.

**Comment by dantjones** [comment]
Closed Issue #17 by adding a series of prechecks to increase the likelihood that Create will succeed. There are still situations beyond our ability to catch in a precheck that could cause the create to fail. THe user will need to do manual clean-up in this case. This may include needing to delete the Docker Machine created directory for the machine including the machine's config.json. This directory should show up under ~/.docker/machine/machines/
